File: python-algorithm/ds_3_hash/L387.py
# -*- coding: utf-8 -*-
# @Time    : 2020/10/1 21:21
# @Author  : Heng Li
# @File    : L387.py
# @Software: PyCharm

class Solution:
    def firstUniqChar(self, s: str) -> int:
        d =  {}
        for c in s:
            if c not in d:
                d[c] = 1
            else:
                d[c] += 1
        for i in range(len(s)):
            if d[s[i]] == 1:
                return i
        return -1

File: python-algorithm/ds_3_hash/L387.cpp

#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> record(26, 0);
        for(int i=0; i<s.size(); ++i)
            record[s[i]-'a']++;
        for(int i=0; i<s.size(); ++i)
            if(record[s[i]-'a'] == 1)
                return i;
        return -1;
    }
};